pkg/cmd/ssh_openstack.go

```diff
@@ -89,11 +88,15 @@ func sshToOpenstackNode(nodeName, path, user, pathSSKeypair string, sshPublicKey
 	cmd.Stderr = os.Stderr
 	if err := cmd.Run(); err != nil {
 		fmt.Println(err)
+		if exitError, ok := err.(*exec.ExitError); ok {
+			sshStatus = exitError.ExitCode()
+		}
 	}
+	defer a.cleanUpOpenstack(sshStatus)
```

Comment:
what's the purpose of having `defer` as last statement of the function? I would refrain from moving the deferred cleanup call to the bottom.

Reply:
during my test, if i move the `defer a.cleanUpOpenstack(sshStatus)` to above lines of value assignment of `sshStatus `, it won't get correct values.
Say if i keep `defer a.cleanUpOpenstack(sshStatus)`  to original line, when some error occurs in ssh, `sshStatus` will be assigned with value 1 (for example) but in `defer a.cleanUpOpenstack(sshStatus)` the value of `sshStatus` is still `0`
do you have a better idea how i can get `sshStatus` correctly passed to `cleanUpOpenstack` meanwhile i can keep the original line for the defer function? thanks!

Reply:
if you want to achieve this you need to pass a pointer

Reply:
thanks! i will have a try

Reply:
Hi @petersutter , i've modified the code using *int to pass values, thanks for your suggestion!
-Neo